### DISCUSSION TITLE
Any guides for deploying to PythonAnywhere?

### DISCUSSION THREAD

**Discussion by Jefficus** [posativ/isso#948] [Q&A] [answered]
I've got isso running in my lab and now I want to deploy it to a free account on PythonAnywhere for further testing. But I'm not really a back-end guy, so I don't know enough about the plumbing to fill in their WepApp deployment form, and Google isn't turning up any obvious help, so I thought I'd ask here. Has anybody written a guide to deploying on PA that I'm just not finding?

**Comment by Jefficus** [comment, marked as answer]
After chasing my tail for a full day, digging into server logs, config files, and source code, I've finally got a simple setup that seems to be working. So in the very great likelihood that I'll completely forget how I did it and be asking the same question a month from now, I'll document the important bits here.

1. Create the PA webapp and choose Custom configuration.
2. Launch a bash shell on PA
3. Create a virtualenv. (I called mine: venv-isso)
4. Activate the virtual env: `source venv-isso/bin/activate`
5. Install isso: `pip install isso`
6. At top level of my user files (/home/USERNAME/) create isso.cfg setting whatever behavioral options you want.
7. The important bit was to give dbpath a local file path to create and manage, like maybe /home/USERNAME/isso_commentary.db
8. Also make sure host is set to a list of URLs that your website might be known as when it calls the comment API. Mine looks like:
```
    host = 
        http://domain.name
        https://domain.name
        http://www.domain.name
        https://www.domain.name
```
9. Save and close the isso.cfg
10. PA will have created a template wsgi.py file for you. Mine was at: /var/www/username_pythonanywhere_com_wsgi.py
11. Edit that file and make it look something like this:
```
import site
site.addsitedir("/path/to/your/venv")
import os
from isso import make_app
from isso import dist, config
application = make_app(
                config.load( config.default_file(),
                             "/path/to/your/isso.cfg"),
                multiprocessing=True)
```

And that's it. I've probably overlooked things, and I'm sure people who actually know what they're doing will find lots that can be improved here, but this should get you pretty close. Reload your webapp and try refreshing your website page to see if it starts including all your new commenty goodness.

If not, I found the PA error and server logs helpful, as well as viewing the console output from the Chrome tab that tried to load your site. Most of my work figuring this out was simply making the errors in those windows go away. :-)

**Reply by ix5** [reply]
Very well written, Jeff, good that you figured it out yourself.

Since others might find themselves in a similar predicament, you might also add this piece of information to the docs, or at least link to this page from the docs.

**Comment by ogurec-ogurec** [comment]
Hi, @Jefficus could you help me install isso-comments on Django (windows) on my local computer?

**Reply by Jefficus** [reply]
I wish my memory was good enough to be able to help, but I don't even remember writing that, and I know nothing about Django or Windows. Sorry.
